src/securejoin.rs

```diff
@@ -616,12 +616,17 @@ pub(crate) async fn observe_securejoin_on_other_device(
                 peerstate.prefer_encrypt = EncryptPreference::Mutual;
                 peerstate.save_to_db(&context.sql).await.unwrap_or_default();
 
-                ChatId::set_protection_for_contact(
-                    context,
-                    contact_id,
-                    mime_message.timestamp_sent,
-                )
-                .await?;
+                if context
+                    .get_config_bool(Config::VerifiedOneOnOneChats)
+                    .await?
+                {
+                    ChatId::set_protection_for_contact(
+                        context,
+                        contact_id,
+                        mime_message.timestamp_sent,
+                    )
+                    .await?;
+                }
```

Comment:
Moving the discussion here from https://github.com/deltachat/deltachat-core-rust/pull/5094#discussion_r1423839766 (message by @Hocuri):
> The protected status of chats is still maintained even if `Config::VerifiedOneOnOneChats` is not set, so it was correct not to check for `Config::VerifiedOneOnOneChats`. `Config::VerifiedOneOnOneChats` only means that the UI has support for it (in hindsight, this wasn't a good name for this config). See
> 
> https://github.com/deltachat/deltachat-core-rust/blob/582541634f806c3044d41cc54e6b0d0e6f8cb081/src/receive_imf.rs#L801-L824

This call to `ChatId::set_protection_for_contact()` isn't related to the chat that is the subject of the Securejoin, it only sets the protection for the 1:1 chat with the verified contact, also adding a protection message to the chat. But if the UI doesn't support "verified 1:1 chats" and these protection messages, how should it display them? So, i just made the logic the same as in `receive_imf.rs` and also in `secure_connection_established()` here -- if the config option is unset, don't add protection to the 1:1 chat, but only remove it if needed (but here we only add it). Maybe you mean that we should always maintain the 1:1 chat protection state (regardless of the config setting), but add protection messages only if the config option is set?

Reply:
> But if the UI doesn't support "verified 1:1 chats" and these protection messages, how should it display them?

As regular info messages

> So, i just made the logic the same as in `receive_imf.rs` and also in `secure_connection_established()` here

Ah yes, the logic there is wrong. Which doesn't matter that much, sooner or later the chat will be marked as protected.

> but then it's a question what a non-supporting UI should do if a chat goes to the "protection broken" state.

If there UI doesn't support verified 1:1 chats, the "protection broken" state is skipped and the chat goes right into unprotected without any explicit confirmation from the user.

----

Although of course what I'm basing my comment on are just the semantics _I thought of_ when implementing verified 1:1 chats. We can change them, and say that if the UI doesn't support verified 1:1 chats, no 1:1 chats are marked as verified.

It doesn't matter that much what happens for UIs that don't support verified 1:1 chats, as long as things somewhat work - we even wanted to remove this setting at some point, after all UIs support them.

Reply:
> As regular info messages.

I also vote for this. Then let's add these messages everywhere (along with marking a chat protected), in `receive_imf.rs` as well f.e.

> If there UI doesn't support verified 1:1 chats, the "protection broken" state is skipped and the chat goes right into unprotected without any explicit confirmation from the user.

+1, looks like the best choice for non-supporting UIs.

> Although of course what I'm basing my comment on are just the semantics _I thought of_ when implementing verified 1:1 chats. We can change them, and say that if the UI doesn't support verified 1:1 chats, no 1:1 chats are marked as verified.

I'd prefer they to be marked as verified, this info may be useful when an UI is updated to a version supporting verified 1:1 chats.

> It doesn't matter that much what happens for UIs that don't support verified 1:1 chats, as long as things somewhat work - we even wanted to remove this setting at some point, after all UIs support them.

Yes, maybe it doesn't matter much, but now the logic differs in different places and it looks like a bug, maybe minor but still. So, let me bring the code into line with your original idea then.

Reply:
I'm wrong -- in `receive_imf.rs` the logic is already correct then (for non-supporting UIs all things are done except marking chats as protection-broken). It's not correct in `securejoin.rs:secure_connection_established()` f.e. I'll check other places and fix them if needed